Query param rigour

smulube opened the pull request.

peteraba commented:
@smulube can you tell me why this PR was closed? seems to do what I need.

peteraba commented:
never mind, I solved the problem by leaving out query parameters and defining a custom Responder.

smulube commented:
Oh hey @peteraba - I actually can't remember now; we actually ended up maintaining our own fork which ended up diverging quite a bit from this one (https://github.com/thingful/httpmock); and purely out of an embarrassing excess of reticence I never actually tried to submit all the changes back upstream.

In fact most recently, we took that fork and renamed and this is what we are currently using internally - https://github.com/thingful/simular. The API is still very similar to this repo; the notable changes were just a slightly cleaner API for registering multiple stubs, and we also added a way of verifying that all stubs were actually called which we call at the end of a single test.
